clients/pkg/promtail/client/client.go

```diff
@@ -410,6 +410,17 @@ func (c *client) send(ctx context.Context, tenantID string, buf []byte) (int, er
 		req.Header.Set("X-Scope-OrgID", tenantID)
 	}
 
+	// Add custom headers on request
+	if len(c.cfg.Headers) > 0 {
+		for k, v := range c.cfg.Headers {
+			if req.Header.Get(k) == "" {
+				req.Header.Add(k, v)
+			}else {
```

Comment:
linter may fail here?

```suggestion
			} else {
```